### ISSUE TITLE
Roblox is subsetting api.roblox.com

### ISSUE THREAD

**Issue by blake-mealey** [blake-mealey/mantle#144]
Context: https://devforum.roblox.com/t/action-needed-sunsetting-apirobloxcom/1796897

Affected endpoints we currently use:

- [x] GET /marketplace/game-pass-product-info -> A new endpoint is not ready at the time of this announcement. Please stay tuned.
- [x] GET /developerproducts/list -> [Develop Api](https://develop.roblox.com/docs#!/DeveloperProducts/get_v1_universes_universeId_developerproducts)
- [x] POST /universes/create -> A new endpoint is not ready at the time of this announcement. Please stay tuned.

Currently none of these endpoints actually has a replacement we can use. They linked to Develop Api for one but there is no analogous API there.

**Comment by blake-mealey** [comment]
Update: looks like [someone else](https://devforum.roblox.com/t/action-needed-sunsetting-apirobloxcom/1796897/8?u=m0rgoth) noticed the developerproducts API issue. There is an API but it doesn't return the necessary data.

**Comment by blake-mealey** [comment]
Update: [roblox has further announced](https://devforum.roblox.com/t/action-needed-sunsetting-apirobloxcom/1796897/36?u=m0rgoth) the following:

* They will release a replacement API at least 1 month before removing the `POST /universes/create` endpoint
* They acknowledged the developerproducts API issue and will "circle back"
* They will release a replacement API at least 1 month before removing the `GET /marketplace/productinfo` endpoint which is similar to the one we use

All in all it sounds like we should be getting proper replacements for all of these with a reasonable amount of time to switch.

**Comment by Julli4n** [comment]
> * GET /marketplace/game-pass-product-info -> A new endpoint is not ready at the time of this announcement. Please stay tuned.

You *could* just use https://economy.roblox.com/v2/assets/1818/details as it's been used in Studio for the past year, and they're likely to announce this anyways.



> Update: looks like [someone else](https://devforum.roblox.com/t/action-needed-sunsetting-apirobloxcom/1796897/8?u=m0rgoth) noticed the developerproducts API issue. There is an API but it doesn't return the necessary data.

They just recently created a new API (undocumented) to get a productId from a developerProductId: https://develop.roblox.com/v1/developerproducts/12973870

**Comment by blake-mealey** [comment]
Thanks for that info!

**Comment by blake-mealey** [comment]
Looked into this some more. Using the economy API to replace /marketplace/game-pass-product-info won't work because it doesn't work for game passes unfortunately. For example, for the product returned from https://games.roblox.com/v1/games/3084813470/game-passes, neither the `id` nor `productId` return the game pass's info:

* https://economy.roblox.com/v2/assets/24924423/details
* https://economy.roblox.com/v2/assets/1221337967/details

For the developer products, the new endpoint could be used to replace our need for the list endpoint during product creation, but we would still need an analogous API for importing.

**Comment by Julli4n** [comment]
> Looked into this some more. Using the economy API to replace /marketplace/game-pass-product-info won't work because it doesn't work for game passes unfortunately. For example, for the product returned from https://games.roblox.com/v1/games/3084813470/game-passes, neither the `id` nor `productId` return the game

For details by product ID: https://economy.roblox.com/v2/developer-products/1/details
For details by pass ID: https://economy.roblox.com/v1/game-pass/1/game-pass-product-info

> * POST /universes/create -> A new endpoint is not ready at the time of this announcement. Please stay tuned.
Use https://apis.roblox.com/universes/v1/universes/create?groupId=GROUPID
Request Body:
```json
{
    "templatePlaceId": 95206881
}```

Response body:
```json
{
    "rootPlaceId": number,
    "universeId": number
}
```

**Comment by Julli4n** [comment]
Additionally, for a direct replacement for `/developerproducts/list`:
`https://apis.roblox.com/developer-products/v1/developer-products/list?universeId=13058` Accepts the same params and sends the same response format.

**Comment by blake-mealey** [comment]
All of these have been resolved now in the latest version: 0.11.6

Thanks @Julli4n for your help with this! Very much appreciated.
